packages/cli-kit/src/public/node/archiver.ts

```diff
@@ -27,10 +49,30 @@ export async function zip(inputDirectory: string, outputZipPath: string): Promis
 
     for (const filePath of pathsToZip) {
       const fileRelativePath = relativePath(inputDirectory, filePath)
-      archive.file(filePath, {name: fileRelativePath})
+
+      if (usePlatformCompatiblePath) {
+        archive.file(platformCompatiblePath(filePath), {name: platformCompatiblePath(fileRelativePath)})
+      } else {
+        archive.file(filePath, {name: fileRelativePath})
+      }
     }
 
     // eslint-disable-next-line @typescript-eslint/no-floating-promises
     archive.finalize()
   })
 }
+
+/**
+ * Takes forward slash file path as a input and returns backslash
+ * file path if platform is Windows.
+ *
+ * @param filePath - A file path.
+ * @returns A file path compatible with the current platform architecture.
+ */
+function platformCompatiblePath(filePath: string) {
+  if (platform === 'windows') {
+    return filePath.replaceAll('/', '\\')
+  }
+
+  return filePath
+}
```

Comment:
Maybe this function makes more sense to move it to `path.ts`, what do you think?